pkg/porter/version.go

```diff
@@ -1,80 +1,111 @@
 package porter
 
 import (
+	"bytes"
 	"fmt"
 	"github.com/deislabs/porter/pkg"
+	"github.com/deislabs/porter/pkg/context"
 	"github.com/deislabs/porter/pkg/mixin"
 	"github.com/deislabs/porter/pkg/porter/version"
 	"github.com/deislabs/porter/pkg/printer"
 	"github.com/pkg/errors"
-	"io"
 	"runtime"
+	"text/template"
 )
 
+type VersionOpts struct {
+	version.Options
+	System bool
+}
+
 type SystemInfo struct {
-	OS string
+	OS   string
 	Arch string
 }
 
-func (p *Porter) PrintVersion(opts version.Options) error {
+type Mixins []mixin.Metadata
+
+type SystemDebugInfo struct {
+	Version mixin.Metadata   `json:"version"`
+	SysInfo SystemInfo       `json:"system"`
+	Mixins  Mixins `json:"mixins"`
+}
+
+func (mixins Mixins) PrintMixinsTable() string {
+	buffer := &bytes.Buffer{}
+	printMixinRow :=
+		func(v interface{}) []interface{} {
+			m, ok := v.(mixin.Metadata)
+			if !ok {
+				return nil
+			}
+			return []interface{}{m.Name, m.VersionInfo.Version, m.VersionInfo.Author}
+		}
+	err := printer.PrintTable(buffer, mixins, printMixinRow, "Name", "Version", "Author")
+	if err != nil {
+		return ""
+	}
+	return buffer.String()
+}
+
+func (p *Porter) PrintVersion(opts VersionOpts) error {
 	metadata := mixin.Metadata{
 		Name: "porter",
 		VersionInfo: mixin.VersionInfo{
 			Version: pkg.Version,
 			Commit:  pkg.Commit,
 		},
 	}
-	return version.PrintVersion(p.Context, opts, metadata)
-}
 
-func getSystemInfo() *SystemInfo {
-	return &SystemInfo{
-		OS:            runtime.GOOS,
-		Arch:          runtime.GOARCH,
+	if opts.System {
+		return p.PrintDebugInfo(p.Context, opts, metadata)
 	}
-}
 
-func (system *SystemInfo) printSystemInfo(out io.Writer) error {
-	_, err := fmt.Fprintf(out, "os: %s\n", system.OS)
-	_, err = fmt.Fprintf(out, "arch: %s\n", system.Arch)
-	return err
+	return version.PrintVersion(p.Context, opts.Options, metadata)
 }
 
-func printSectionHeader(out io.Writer, header string){
-	underline := "-------"
-	_, _ = fmt.Fprintf(out, "\n%s\n%s\n", header, underline)
-
+func getSystemInfo() *SystemInfo {
+	return &SystemInfo{
+		OS:   runtime.GOOS,
+		Arch: runtime.GOARCH,
+	}
 }
 
-func (p *Porter) PrintDebugInfo(opts version.Options) error {
-	// force opts to print version as plaintext
+func (p *Porter) PrintDebugInfo(ctx *context.Context, opts VersionOpts, versionMetadata mixin.Metadata) error {
 	opts.RawFormat = string(printer.FormatPlaintext)
-	err := p.PrintVersion(opts)
+	mixins, err := p.ListMixins()
+	sysInfo := getSystemInfo()
 	if err != nil {
-		return errors.Wrap(err, "Failed to print version")
+		_ = errors.Wrap(err, "Failed to get list of mixins")
```

Comment:
Instead of ignoring the error (like it's doing now), or returning it (which would make it very hard to use this command for troubleshooting problems 😅 ), we can have it print a debug message and then keep going:

The contributing guide has a section on how to handle debug logging:

https://github.com/deislabs/porter/blob/master/CONTRIBUTING.md#logging

You can check `p.Debug` and then print to `p.Err` following the example in the guide, then have the rest of the command continue. That way even if we can't print any mixin data, at least we are able to print the data we _do_ have.